Recent macOS builds fail to run

Automated macOS builds currently fail to run, due to a signature validation error when loading libfftw3f.3.dylib. This library comes from Homebrew: https://formulae.brew.sh/formula/fftw

The problem began when the fftw bottle was updated to 3.3.10_2. `otool -l` shows that the library's SDK version is set to "n/a", but it needs to be set to 10.9 or later for the corresponding signature to be accepted.

Homebrew is tracking the issue here: https://github.com/Homebrew/homebrew-core/issues/209981

For now, I guess we'll have to wait until the issue is fixed and an updated bottle is published.

this is independant of gqrx bundling:
```
SoapySDRUtil --probe=driver=rtlsdr
######################################################
##     Soapy SDR -- the SDR abstraction library     ##
######################################################

Probe device driver=rtlsdr
[ERROR] SoapySDR::loadModule(/Users/kai/radioconda/lib/SoapySDR/modules0.8/librtlsdrSupport.so)
  dlopen() failed: dlopen(/Users/kai/radioconda/lib/SoapySDR/modules0.8/librtlsdrSupport.so, 0x0001): Library not loaded: @rpath/librtlsdr.2.dylib
  Referenced from: <D19B0E5C-E821-3318-B4D4-7EE2263BA815> /Users/kai/radioconda/lib/SoapySDR/modules0.8/librtlsdrSupport.so
  Reason: tried: '/Users/kai/radioconda/lib/SoapySDR/modules0.8/../../librtlsdr.2.dylib' (no such file), '/Users/kai/radioconda/lib/SoapySDR/modules0.8/../../librtlsdr.2.dylib' (no such file), '/Users/kai/radioconda/lib/librtlsdr.2.dylib' (no such file), '/Users/kai/radioconda/bin/../lib/librtlsdr.2.dylib' (no such file), '/usr/local/lib/librtlsdr.2.dylib' (no such file), '/usr/lib/librtlsdr.2.dylib' (no such file, not in dyld cache)
Error probing device: SoapySDR::Device::make() no match
```

It works for me, macos 15.5, mac mini m2 pro and macbook air m2.
I tried airspyhf with (soapy=0,driver=airspyhf) and without soapy.

I get some soapy errors from the console:
```
./Gqrx.app/Contents/MacOS/gqrx
gr-osmosdr 0.2.0.0 (0.2.0) gnuradio 3.10.12.0
built-in source types: file rtl rtl_tcp uhd miri hackrf bladerf rfspace airspy airspyhf soapy redpitaya 
Resampling audio 96000 -> 48000
PortAudio V19.7.0-devel, revision 147dd722548358763a8b649b3e4b41dfffbcfbb6 (version 1246976)
Number of audio devices: 10
  0:  WaveShare  I:0  O:2
  1:  Studio Display Microphone  I:1  O:0
  2:  Studio Display Speakers  I:0  O:8
  3:  USB AUDIO  CODEC  I:0  O:2
  4:  USB AUDIO  CODEC  I:2  O:0
  5:  UAC3556B  I:2  O:2
  6:  Jabra Speak 710  I:0  O:2
  7:  Jabra Speak 710  I:1  O:0
  8:  Mac mini-Lautsprecher  I:0  O:2
  9:  Microsoft Teams Audio  I:1  O:1
Using default audio device
BandPlanFile is /Users/kai/.config/gqrx/bandplan.csv
BookmarksFile is /Users/kai/.config/gqrx/bookmarks.csv
PortAudio V19.7.0-devel, revision 147dd722548358763a8b649b3e4b41dfffbcfbb6 (version 1246976)
Number of audio devices: 10
  0:  WaveShare  I:0  O:2
  1:  Studio Display Microphone  I:1  O:0
  2:  Studio Display Speakers  I:0  O:8
  3:  USB AUDIO  CODEC  I:0  O:2
  4:  USB AUDIO  CODEC  I:2  O:0
  5:  UAC3556B  I:2  O:2
  6:  Jabra Speak 710  I:0  O:2
  7:  Jabra Speak 710  I:1  O:0
  8:  Mac mini-Lautsprecher  I:0  O:2
  9:  Microsoft Teams Audio  I:1  O:1
[INFO] [UHD] Mac OS; Clang version 18.1.8 ; Boost_108600; UHD_4.8.0.0-release
[ERROR] SoapySDR::loadModule(/Users/kai/Desktop/Gqrx.app/Contents/soapy-modules/libLMS7Support.so)
  duplicate entry for lime (/Users/kai/radioconda/lib/SoapySDR/modules0.8/libLMS7Support.so)
[ERROR] SoapySDR::loadModule(/Users/kai/Desktop/Gqrx.app/Contents/soapy-modules/libPlutoSDRSupport.so)
  duplicate entry for plutosdr (/Users/kai/radioconda/lib/SoapySDR/modules0.8/libPlutoSDRSupport.so)
[ERROR] SoapySDR::loadModule(/Users/kai/Desktop/Gqrx.app/Contents/soapy-modules/libaudioSupport.so)
  duplicate entry for audio (/Users/kai/radioconda/lib/SoapySDR/modules0.8/libaudioSupport.so)
[ERROR] SoapySDR::loadModule(/Users/kai/Desktop/Gqrx.app/Contents/soapy-modules/libremoteSupport.so)
  duplicate entry for remote (/Users/kai/radioconda/lib/SoapySDR/modules0.8/libremoteSupport.so)
[ERROR] SoapySDR::ConverterRegistry(S8, S16, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(S8, F32, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(S8, F64, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(S16, S8, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(S16, F32, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(S16, F64, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(S32, F32, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(S32, F64, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(F32, S8, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(F32, S16, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(F32, S32, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(F32, F32, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(F32, F64, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(F64, S8, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(F64, S16, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(F64, S32, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(F64, F32, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(CS8, CS16, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(CS8, CF32, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(CS8, CF64, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(CS16, CS8, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(CS16, CF32, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(CS16, CF64, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(CS32, CF32, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(CS32, CF64, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(CF32, CS8, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(CF32, CS16, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(CF32, CS32, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(CF32, CF32, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(CF32, CF64, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(CF64, CS8, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(CF64, CS16, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(CF64, CS32, 3) duplicate registration
[ERROR] SoapySDR::ConverterRegistry(CF64, CF32, 3) duplicate registration
[WARNING] Unable to scan local: -19

[WARNING] SoapySSDPEndpoint failed join group udp://[ff02::c]:1900 on en1
  setsockopt(IPV6_MULTICAST_IF, fd11:7e23:4266:0:1c2f:a223:b650:608e) [22: Invalid argument]
[WARNING] SoapySSDPEndpoint failed join group udp://[ff02::c]:1900 on en1
  setsockopt(IPV6_MULTICAST_IF, 2003:e5:c717:e200:8bb:77ec:cd16:e922) [22: Invalid argument]
[WARNING] SoapySSDPEndpoint failed join group udp://[ff02::c]:1900 on en1
  setsockopt(IPV6_MULTICAST_IF, 2003:e5:c717:e200:3ddd:387e:abc2:7ea) [22: Invalid argument]
[WARNING] SoapySSDPEndpoint failed join group udp://[ff02::c]:1900 on en1
  setsockopt(IPV6_MULTICAST_IF, 2003:e5:c71e:b400:dc:9021:f3c5:3c39) [22: Invalid argument]
[WARNING] SoapySSDPEndpoint failed join group udp://[ff02::c]:1900 on en1
  setsockopt(IPV6_MULTICAST_IF, 2003:e5:c71e:b400:e8a2:9cc3:25e2:bda4) [22: Invalid argument]
[WARNING] SoapySSDPEndpoint failed join group udp://[ff02::c]:1900 on utun4
  setsockopt(IPV6_MULTICAST_IF, fd7a:115c:a1e0::a501:6337) [22: Invalid argument]
gr-osmosdr 0.2.0.0 (0.2.0) gnuradio 3.10.12.0
built-in source types: file rtl rtl_tcp uhd miri hackrf bladerf rfspace airspy airspyhf soapy redpitaya 
PortAudio V19.7.0-devel, revision 147dd722548358763a8b649b3e4b41dfffbcfbb6 (version 1246976)
Number of audio devices: 10
  0:  WaveShare  I:0  O:2
  1:  Studio Display Microphone  I:1  O:0
  2:  Studio Display Speakers  I:0  O:8
  3:  USB AUDIO  CODEC  I:0  O:2
  4:  USB AUDIO  CODEC  I:2  O:0
  5:  UAC3556B  I:2  O:2
  6:  Jabra Speak 710  I:0  O:2
  7:  Jabra Speak 710  I:1  O:0
  8:  Mac mini-Lautsprecher  I:0  O:2
  9:  Microsoft Teams Audio  I:1  O:1
Using default audio device
Decimation: 4
  stage: 1  ratio: 4
Oportaudio_sink::work(): Error writing to audio device: Output underflowed
portaudio_sink::work(): Error writing to audio device: Output underflowed
portaudio_sink::work(): Error writing to audio device: Output underflowed
'''

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -85,71 +85,28 @@ jobs:
           else
             echo "available=false" >> $GITHUB_OUTPUT;
           fi
-      - name: Install dependencies
-        run: |
-          uname -a
-          # for https://github.com/actions/runner-images/issues/9272
-          sudo chown -R runner:admin /usr/local/
-          brew update
-          brew install --HEAD librtlsdr
-          brew install airspy airspyhf boost dylibbundler gnuradio hackrf libbladerf libserialport portaudio pybind11 six soapyremote uhd qt@6 || true
-
-          cd /tmp
-          git clone https://github.com/analogdevicesinc/libiio.git
-          cd libiio
-          git checkout v0.26
-          mkdir build
-          cd build
-          cmake -DCMAKE_BUILD_TYPE=Release ..
-          make -j4
-          sudo make install
-
-          cd /tmp
-          git clone https://github.com/analogdevicesinc/libad9361-iio.git
-          cd libad9361-iio
-          mkdir build
-          cd build
-          cmake -DCMAKE_POLICY_VERSION_MINIMUM=3.5 -DCMAKE_BUILD_TYPE=Release ..
-          make -j4
-          sudo make install
-
-          cd /tmp
-          git clone https://github.com/pothosware/SoapyPlutoSDR.git
-          cd SoapyPlutoSDR
-          mkdir build
-          cd build
-          cmake -DCMAKE_BUILD_TYPE=Release ..
-          make -j4
-          sudo make install
-
-          cd /tmp
-          sudo cp /Library/Frameworks/iio.framework/iio /usr/local/lib/libiio.dylib
-          sudo install_name_tool -id "/usr/local/lib/libiio.dylib" /usr/local/lib/libiio.dylib
-          sudo cp /Library/Frameworks/ad9361.framework/ad9361 /usr/local/lib/libad9361.dylib
-          sudo install_name_tool -id "/usr/local/lib/libad9361.dylib" /usr/local/lib/libad9361.dylib
-          sudo install_name_tool -delete_rpath /Library/Frameworks /usr/local/lib/libad9361.dylib
-          sudo install_name_tool -change @rpath/iio.framework/Versions/0.23/iio /usr/local/lib/libiio.dylib /usr/local/lib/libad9361.dylib
-          sudo install_name_tool -change @rpath/iio.framework/Versions/0.23/iio /usr/local/lib/libiio.dylib /usr/local/lib/SoapySDR/modules0.*/libPlutoSDRSupport.so
-          sudo install_name_tool -change @rpath/ad9361.framework/Versions/0.2/ad9361 /usr/local/lib/libad9361.dylib /usr/local/lib/SoapySDR/modules0.*/libPlutoSDRSupport.so
-
-          cd /tmp
-          git clone https://gitea.osmocom.org/sdr/gr-iqbal.git
-          cd gr-iqbal
-          git submodule update --init --recursive
-          mkdir build
-          cd build
-          cmake -DCMAKE_BUILD_TYPE=Release ..
-          make -j4
-          sudo make install
-
-          cd /tmp
-          git clone https://gitea.osmocom.org/sdr/gr-osmosdr.git
-          cd gr-osmosdr
-          mkdir build
-          cd build
-          cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_FLAGS=-Wno-register ..
-          LIBRARY_PATH=/usr/local/opt/icu4c/lib make -j4
-          sudo make install
+      - name: Install conda dependencies
+        id: setup-micromamba
+        uses: mamba-org/setup-micromamba@v2
+        with:
+          environment-name: gqrx
+          create-args: >-
+            c-compiler
+            cxx-compiler
+            cmake
+            make
+            pkg-config
+            gnuradio-core
+            gnuradio-osmosdr
+            libboost-devel
+            qt6-main
+            soapysdr
+            soapysdr-module-audio
+            soapysdr-module-lms7
+            soapysdr-module-plutosdr
+            soapysdr-module-remote
+            soapysdr-module-volk-converters
+            volk
       - name: Install Apple certificate
         if: ${{ steps.secret-check.outputs.available == 'true' }}
         env:
@@ -173,11 +130,14 @@ jobs:
         with:
           fetch-depth: 0
       - name: Configure
+        shell: bash -el {0}
         run: mkdir build && cd build && cmake -DCMAKE_BUILD_TYPE=Release ..
       - name: Compile
         working-directory: build
         run: make -j4
       - name: Build app bundle
+        env:
+          CONDA_PREFIX: ${{ steps.setup-micromamba.outputs.environment-path }}
         run: ./macos_bundle.sh ${{ steps.secret-check.outputs.available }}
       - name: Notarize app bundle
         if: ${{ steps.secret-check.outputs.available == 'true' }}

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -72,24 +72,26 @@ jobs:
         backend: [Portaudio, Gr-audio]
     runs-on: ${{ matrix.os }}
     steps:
-      - name: Install dependencies
-        run: |
-          # for https://github.com/actions/runner-images/issues/9272
-          sudo chown -R runner:admin /usr/local/
-          brew update
-          brew install airspy boost gnuradio hackrf libbladerf librtlsdr pybind11 six uhd qt@6 || true
-
-          cd /tmp
-          git clone https://gitea.osmocom.org/sdr/gr-osmosdr.git
-          cd gr-osmosdr
-          mkdir build
-          cd build
-          cmake -DCMAKE_CXX_FLAGS=-Wno-register ..
-          LIBRARY_PATH=/usr/local/opt/icu4c/lib make -j4
-          sudo make install
+      - name: Install conda dependencies
+        id: setup-micromamba
+        uses: mamba-org/setup-micromamba@v2
+        with:
+          environment-name: gqrx
+          create-args: >-
+            c-compiler
+            cxx-compiler
+            cmake
+            make
+            pkg-config
+            gnuradio-core
+            gnuradio-osmosdr
+            libboost-devel
+            qt6-main
+            volk
       - name: Checkout code
         uses: actions/checkout@v4
       - name: Configure
+        shell: bash -el {0}
         run: mkdir build && cd build && cmake -DOSX_AUDIO_BACKEND:STRING=${{ matrix.backend }} ..
       - name: Compile
         working-directory: build

--- a/macos_bundle.sh
+++ b/macos_bundle.sh
@@ -2,8 +2,16 @@
 
 GQRX_VERSION="$(<build/version.txt)"
 IDENTITY=Y3GC27WZ4S
-BREW_PREFIX="$(brew --prefix)"
-MACDEPLOYQT6="${BREW_PREFIX}"/opt/qt@6/bin/macdeployqt
+
+echo "CONDA_PREFIX: " $CONDA_PREFIX
+
+MACDEPLOYQT6=${CONDA_PREFIX}/bin/macdeployqt6
+echo "macdeployqt6: " ${MACDEPLOYQT6}
+
+# cleanup and setup
+if [ -e Gqrx.app ] ;
+then rm -r Gqrx.app
+fi
 
 mkdir -p Gqrx.app/Contents/MacOS
 mkdir -p Gqrx.app/Contents/Resources
@@ -51,20 +59,15 @@ EOM
 
 cp build/src/gqrx Gqrx.app/Contents/MacOS
 cp resources/icons/gqrx.icns Gqrx.app/Contents/Resources
-# NOTE: PlutoSDR is built locally, so it will not in the brew path
-cp /usr/local/lib/SoapySDR/modules*/libPlutoSDRSupport.so Gqrx.app/Contents/soapy-modules
-cp "${BREW_PREFIX}"/lib/SoapySDR/modules*/libremoteSupport.so Gqrx.app/Contents/soapy-modules
-chmod 644 Gqrx.app/Contents/soapy-modules/*
+cp "$CONDA_PREFIX"/lib/SoapySDR/modules*/* Gqrx.app/Contents/soapy-modules
 
-dylibbundler -s "${BREW_PREFIX}"/opt/icu4c/lib/ -od -b -x Gqrx.app/Contents/MacOS/gqrx -x Gqrx.app/Contents/soapy-modules/libPlutoSDRSupport.so -x Gqrx.app/Contents/soapy-modules/libremoteSupport.so -d Gqrx.app/Contents/libs/
-"${MACDEPLOYQT6}" Gqrx.app -no-strip -always-overwrite # TODO: Remove macdeployqt workaround
 if [ "$1" = "true" ]; then
-    "${MACDEPLOYQT6}" Gqrx.app -no-strip -always-overwrite -sign-for-notarization="${IDENTITY}"
+    "${MACDEPLOYQT6}" Gqrx.app -verbose=1 -no-strip -always-overwrite -sign-for-notarization="${IDENTITY}" -libpath=Gqrx.app/Contents/Frameworks
 else
-    "${MACDEPLOYQT6}" Gqrx.app -no-strip -always-overwrite
+    "${MACDEPLOYQT6}" Gqrx.app -verbose=1 -no-strip -always-overwrite -libpath=Gqrx.app/Contents/Frameworks
 fi
 
-for f in Gqrx.app/Contents/libs/*.dylib Gqrx.app/Contents/soapy-modules/*.so Gqrx.app/Contents/Frameworks/*.framework Gqrx.app/Contents/Frameworks/*.dylib Gqrx.app/Contents/MacOS/gqrx
+for f in Gqrx.app/Contents/Frameworks/*.dylib Gqrx.app/Contents/soapy-modules/* Gqrx.app/Contents/MacOS/gqrx
 do
     if [ "$1" = "true" ]; then
         codesign --force --verify --verbose --timestamp --options runtime --entitlements /tmp/Entitlements.plist --sign "${IDENTITY}" "$f"

--- a/resources/news.txt
+++ b/resources/news.txt
@@ -3,6 +3,7 @@
 
        NEW: Start/stop I/Q recording via remote control.
   IMPROVED: Allow multiple simultaneous remote control connections.
+     FIXED: DMG release for ARM-based Macs fails to start.
      FIXED: Compilation error affecting MSVC.
 
 

--- a/src/applications/gqrx/main.cpp
+++ b/src/applications/gqrx/main.cpp
@@ -67,6 +67,7 @@ int main(int argc, char *argv[])
     {
         qputenv("SOAPY_SDR_PLUGIN_PATH", plugin_path.toUtf8());
         qputenv("SOAPY_SDR_ROOT", "/invalid");
+        qputenv("CONDA_PREFIX", "/invalid");
     }
 
     // setup controlport via environment variables

